lib/containers/utils.pm

```diff
@@ -149,14 +149,29 @@ sub basic_container_tests {
 
 # Setup environment
 sub set_up {
-    my $dir = shift;
+    my ($dir, $file) = @_;
+    $file //= 'Dockerfile';
+    record_info "$file";
     die "You must define the directory!" unless $dir;
     assert_script_run("mkdir -p $dir/BuildTest");
     assert_script_run "curl -f -v " . data_url('containers/app.py') . " > $dir/BuildTest/app.py";
-    assert_script_run "curl -f -v " . data_url('containers/Dockerfile') . " > $dir/BuildTest/Dockerfile";
+    record_info('Dockerfile', script_output("cat $dir/BuildTest/app.py"));
+    assert_script_run "curl -f -v " . data_url("containers/$file") . " > $dir/BuildTest/Dockerfile";
+    record_info('Dockerfile', script_output("cat $dir/BuildTest/Dockerfile"));
     assert_script_run "curl -f -v " . data_url('containers/requirements.txt') . " > $dir/BuildTest/requirements.txt";
 }
 
+sub build_buildah_container_img {
```

Comment:
Please reuse `build_img` as they're almost the same.

Reply:
there are but i would rather keep it simple and clean. i had enough with the if..else all around. In additional i would have to make extra tests. lets think the design principles later.

Reply:
What is wrong with this?
```perl
sub build_img {
    my $dir = shift;
    die "You must define the directory!" unless $dir;
    my $runtime = shift;
    die "You must define the runtime!" unless $runtime;
    my $registry = get_var('REGISTRY', 'docker.io');
    assert_script_run("cd $dir");
    if ($runtime =~ /docker|podman/) {
        assert_script_run("$runtime image pull $registry/library/python:3", timeout => 300);
        assert_script_run("$runtime tag $registry/library/python:3 python:3");
        assert_script_run("$runtime build -t myapp BuildTest");
    } else {
        assert_script_run("$runtime bud -t myapp BuildTest");
    }
    assert_script_run("$runtime images| grep myapp");
}
```

Reply:
@pdostal nothing wrong with it. but also not necessary. if..else keep spreading!! anyway i applied your suggestion on 946e17650

Reply:
This function can then be removed.

Reply:
right. i forgot to remove it. thatnks